public/javascript/posts/BookAppointmentModal.js

```diff
@@ -46,15 +47,23 @@ angular.module('BookAppointmentApp')
     };
 
     $scope.ok = function () {
-      switch($scope.radioAppointment.selected_type){
-        case 1:
-                $scope.addDoctorInfo();
-                break;
-        case 2:
-                $scope.addPatientInfo();
-                break;
+      if($scope.showPatient && $scope.appointmentForm.$invalid){
```

Comment:
Validations are applied for the window in which doctor creates an appointment for patient. There is no need to add validations on the window where doctor creates appointment for doctor. So ' if ( $scope.showPatient && $scope.appointmentForm.$invalid ) ' is to check if user is creating an appointment for patient and the form validation.